modules/internal/ChapelDomain.chpl

```diff
@@ -1493,6 +1493,46 @@ module ChapelDomain {
       return _newArray(x);
     }
 
+    /*
+      Invoking this method will attempt to create and return an array
+      declared over the domain instance. If there is not enough memory
+      to satisfy the allocation, an error will be thrown, allowing
+      the program to continue if handled, as opposed to halting and
+      thus stopping program execution.
+
+      This method will be most reliable in configurations that use a
+      fixed heap (e.g., when using ``CHPL_GASNET_SEGMENT=large``), but
+      can be called in all configurations. In the case of a dynamic
+      heap, it is possible that overcommit will cause the array
+      allocation to succeed, even if there is not enough physical
+      memory to satisfy the allocation, which will then fail with a bus
+      error when attempting to access the array.
+
+      This method is currently supported on both default rectangular
+      and block domains.
+    */
+    pragma "no copy return"
+    @unstable("tryCreateArray() is subject to change in the future.")
+    proc tryCreateArray(type eltType) throws {
+      if !this.isDefaultRectangular() && !this.isBlock() then
```

Comment:
Instead of adding `isBlock` I think you can do `isSubtype(this.dist.type, Block);` (or something like that, we do a similar check in Arkouda's `contiguousIndices`)